### PR TITLE
Implement the easy level feature

### LINKED ISSUE #32: Implement level selection
Players should be able to select a level other than "Easy". Some ideas:

Harder:
- playing the tones at the same time
- limiting number of times they can replay the tones

Easier:
- give chances to use a "hint" function - e.g. to play the tones between the tones in a scale
- play tones only in an upward direction for easier levels (currently it plays sometimes in a downward direction)

### PR DESCRIPTION
**Pull request by eyeri91**
Closes #32 

In this branch, I have been working on ...

- Add different level buttons in the start page of the game for users to choose among `Easy`, `Intermediate` and `Hard`.
- Disable the `Start` button until user clicks any of level button.
- Pass  which level a user decided to play from `View` to `Model`.
- Work on `Easy` level feature such as ...
     1. Always play notes in upward direction. (ex. C# -> F#)
     2. Add `Get a hint` button to help a beginner user can listen to all notes in scale between 2 random notes between.

Currently I am working on `Easy` level feature. Once it is finished, I will move on to `Intermediate` and eventually `Hard`.

### Add different level buttons

Screen Shot 2022-10-09 at 12 15 03 AM

Screen Shot 2022-10-09 at 12 15 21 AM

I added these buttons temporarily just next to the `Start` button but we can always discuss where they should be. Or whether they should be buttons at all. 😅 

### Share chosen level information between `View` and `Model`
Screen Shot 2022-10-09 at 12 25 41 AM

Screen Shot 2022-10-09 at 12 19 34 AM

Screen Shot 2022-10-09 at 12 19 46 AM

So when it publishes `gameStart` event, I passed the level information in `data` and pass it to the `subscriber`, in this case only to `getNewQuiz()`

Screen Shot 2022-10-09 at 12 44 20 AM

I added this line to avoid error as we only pass `gameLevel` parameter when we start the game, not while it is loading the rest of questions. 

### `Easy` level feature.

Screen Shot 2022-10-09 at 12 33 30 AM

> 1. Always play notes in upward direction. (ex. C# -> F#)

Screen Shot 2022-10-09 at 12 36 08 AM

This function `#getAllNotesWithinScale()` which returns an array of all notes  in case users ask for some hints. 
But I am completely stuck in how to play them in scale. I definitely need your help here.

Also I added `hintButtonCounter = 0;` in `View` for when we want to limit a number of hints users can use.

Thank you :)

### CODE REVIEW

File: src/view.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import * as Tone from 'tone';
 import { toSentenceCase } from './utils';
 
+const levels = ['Easy', 'Intermediate', 'hard'];
```
Comment (kuosandys) [inline src/view.js:4]:
```suggestion
const levels = ['Easy', 'Intermediate', 'Hard'];
```
We uppercase the first letter in our buttons so we won't see this in the UI, but let's keep this consistent 😅

File: src/view.js [hunk 2]
```diff
@@ -30,8 +32,10 @@ export class View {
     publishNewAnswerEvent,
     publishPlayGameAgainEvent
   ) {
+    this.level;
     this.appContainer = musicApp;
     this.isPlayTonesButtonClicked = false;
+    this.hintButtonCounter = 0;
```
Comment (kuosandys) [inline src/view.js:38]: Just a thought - do these need to be public fields? We've talked before about making fields/methods private by default, and only exposing them as public when necessary. This helps to keep the interfaces of our classes simpler, so it's eaiser to for other code to see what it does, and how to work with it. It also helps us limit ways that code outside of the class can modify instances of the class (e.g. setting the level directly). Could we do that here?

**Comment by kuosandys** [comment]
Thanks for workign on this, and nice work so far! Really cool that you're checking and swapping the note orders to ensure that they're always upwards for the easy level, and I like the forward thinking of limiting the number of hints the player can use.

> This function `#getAllNotesWithinScale()` which returns an array of all notes in case users ask for some hints.
> But I am completely stuck in how to play them in scale. I definitely need your help here.

Good work on passing that data to `View`! I see you're trying to play them consecutively, right? This is kind of similar to what we do for our interval buttons (playing one tone after another), and we can also take a look at the [documentation](https://tonejs.github.io/docs/14.7.77/Synth#triggerAttackRelease) for the `triggerAttackAndRelease` method on Tone.js 's `Synth`, specifically the third parameter:
Screen Shot 2022-10-10 at 15 59 03

This optional parameter allows us to pass in a value for a point in time when to play our note. For our interval buttons, we create a variable set to the point in time "now" with `Tone.now()`, then we play the first note at point `now`, and the second note at point `now + 1` (1 second later). There's a bit more explanation for what the `+ 1` part does in the [documentation](https://tonejs.github.io/docs/14.7.77/type/Time) for `Time`.

We could iterate over the `allNotesInScale` array and for each note, play it at an increasingly later time from `now`:
```javascript
  this.#sampler.triggerAttackRelease(note, '8n', now + i);
```
We shouldn't need to use `setTimeout`, since the `triggerAttackAndRelease` method provides us a way to set a time at which to play notes, but that was good intuition!

One extra thing - once you get the tones playing, you'll notice that they start accumulating again, similar to the situation we saw initially with playing intervals 😅 Instead of attaching a new event listener each time we update the question page, we could instead attach it once when we render the question page, and update the `allNotesInScale` data each time we get new question data. This is also similar to how we solved it for the interval buttons.

Hope this explanation was helpful! As always, there are several moving parts to this feature, and I'm always happy to jump in a call to work on it together, or answer any questions that come up 😊

File: src/view.js (hunk 2 above)
Comment (eyeri91) [reply src/view.js:38]: All these don't need to be public actually. I'll update them now!

File: src/view.js [hunk 3]
```diff
@@ -132,8 +161,21 @@ export class View {
...
-      this.#publishNewAnswerEvent(undefined);
+    skipQuestionButton.addEventListener('click', () =>
+      this.#publishNewAnswerEvent(undefined)
+    );
+
+    getHintButton.addEventListener('click', () => {
+      const now = Tone.now();
+
+      for (let i = 0; i < this.#allNotesInCurrentScale.length; i++) {
+        console.log(this.#allNotesInCurrentScale[i]);
```
Comment (kuosandys) [inline src/view.js:172]:
```suggestion
```
Playing notes in a scale works now 🎉 Let's clean this up a bit

**Review by kuosandys — APPROVED** [review]
Tested it out and it works perfectly ❤️ Really awesome that players can now get a hint if they're stuck - besides the small linting suggestion above, I'd say LGTM when you're ready!

Also, if you'd like to update the pull request title to include implementing the easy level features, please feel free 😊

**Pull request merged**